### PR TITLE
Add "bytes" asset type for storing data in the fv3config object

### PR DESCRIPTION
**Pull request by nbren12**
Previously this library could not deploy arbitrary assets to remote file
systems without first storing those assets in a remote storage system (e.g. GCS, Kuberentes config map, etc). This is problematic because 
1. the user must manage these collection of external assets AND this configuration yaml itself. For instance, the configuration yaml will not be valid if this external data is moved or deleted.
1. authentication is needed to work with these remote stores.

These difficulties may be worthwhile for large binary assets (e.g. initial condition netCDFs), but it would be convenient to store some small text files in the configuration dictionary itself. Some examples of files we could use this for include prognostic runfiles, diagtables, serialized scikit-learn models, the nudging_file_list used for the reanalysis nudging workflow, etc.

This PR adds a "bytes" asset type that can save an arbitrary list of
bytes to a specified path. Used in conjuction with patch_files, this
will allow the configuration dict to be in principle a
complete/serializable representation of a run-directory.

Public API added:
- `fv3config.get_bytes_asset_dict`

### CODE REVIEW

File: tests/test_asset_list.py [hunk 1]
```diff
@@ -289,5 +290,14 @@ def test_write_asset(self):
         self.assertTrue(os.path.exists(os.path.join(target_workdir, test_filename)))
 
 
+def test_write_bytes_asset(tmpdir):
+    asset = get_bytes_asset_dict(
+        b"hello world", target_location="", target_name="hello.txt"
+    )
+    write_asset(asset, tmpdir)
+    with tmpdir.join("hello.txt").open("rb") as f:
+        assert f.read() == b"hello world"
```
Comment (mcgibbon) [inline tests/test_asset_list.py:299]: Can you add a test where the asset is being written to a subdirectory? Copy-pasting this test and making a minor change is sufficient.

File: fv3config/cli.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    return parser.parse_args()
+
+
+def write_run_directory():
+    args = _parse_args()
+
+    with fsspec.open(args.config) as f:
+        config = yaml.safe_load(f)
+
+    fv3config.write_run_directory(config, args.rundir)
```
Comment (mcgibbon) [inline fv3config/cli.py]: It feels strange for `fv3config` to define two functions both called `write_run_directory`, could you change the name of this one? You can still keep the same console script name.

**Comment by nbren12** [comment]
Thanks. Yah. I updated the branch and those changes disappeared.

**Comment by nbren12** [comment]
Okay. Good call on the subdirectory test @mcgibbon. That functionality was broken.

**Pull request merged**